libs/pubsub/query/query_test.go

```diff
@@ -51,10 +61,12 @@ func TestMatches(t *testing.T) {
 			require.Nil(t, err)
 		}
 
+		require.NotNil(t, q, "Query '%s' should not be nil", tc.s)
```

Comment:
do we have to check if q is nil? if later it will fail anyway when trying to access q.Matches method.

Reply:
We don't technically have to, `q` may be nil, so I see it as no harm in checking since we can avoid a panic.